modules/users/client/components/AboutMe.component.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import sanitizeHtml from 'sanitize-html';
+import { withNamespaces } from '@/modules/core/client/utils/i18n-angular-load';
+{/* import { Trans } from 'react-i18next'; */}
+import '@/config/lib/i18n';
+import PropTypes from 'prop-types';
+import { limitTo, plainTextLength } from './utils/filters';
+
+export class AboutMe extends Component {
+  constructor(props) {
+    super(props);
+    this.changeProfileDescriptionToggle = this.changeProfileDescriptionToggle.bind(this);
+
+    this.state = {
+      profileDescriptionToggle: false
+    };
+  }
+  changeProfileDescriptionToggle(){
+    this.setState((prevState) => ({
+      profileDescriptionToggle: !prevState.profileDescriptionToggle
+    }));
+  }
+
+  render(){
+    const { t, profile, isSelf, appSettings } = this.props;
+    return (<>
+      <section className="panel panel-default">
+        <header className="panel-heading">
+          {t('About me')}
+        </header>
+        <div className="panel-body">
+          {/* Short descriptions */}
+          {(profile.description.length < 2400 || this.state.profileDescriptionToggle) &&
+          <div dangerouslySetInnerHTML={{__html: sanitizeHtml(profile.description)}}></div>}{/* TODO Can we trust this HTML?*/}
+
+          {/* Long descriptions */}
+          {profile.description.length >= 2400 && !this.state.profileDescriptionToggle &&
+          <div
+            className="panel-more-wrap">
+            <div className="panel-more-excerpt" dangerouslySetInnerHTML={{__html: sanitizeHtml(limitTo(profile.description, 2400))}}></div>
```

Comment:
This splitting the string was always a bit risky because we might split in the middle of an HTML tag or link. :-(

It's okay to merge as such since that's how it was functioning previously.

Just some brainstorming for an alternative approach:

- We're really concerned only about the visual length of this area, so perhaps the toggle `profileDescriptionToggle` could only change a className?
- Length of the string doesn't necessarily correlate with the height of the area (depends on so many things such as width and text styling) so that's probably was the reason why I why I went with string splitting originally. :-(
- This (showing longer text vs and "show more") pattern is used in quite a few places in profile, I wonder if we could wrap it in a component and re-use everywhere?